sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/pubkey-crypto.yaml

```diff
@@ -0,0 +1,151 @@
+provider:
+  metadata:
+    name: Generic public key cryptography
+    category: Cryptography
+    message: Make sure this private key gets revoked, changed, and removed from the code.
+  detection:
+    pre:
+      include:
+        content:
+          - "PRIVATE KEY-----"
+  rules:
+    - id: rsa-private-key
+      metadata:
+        name: RSA private keys should not be disclosed
+      detection:
+        matching:
+          pattern: "-----BEGIN RSA PRIVATE KEY-----"
+          context:
+            matchEach:
+              - patternAfter: "[a-zA-Z0-9+/]{64}" 
```

Comment:
Ok for `{60,}` as stated above.
For `^\\s*` I'm not so sure. It would not match something like:
```python
pkey = "-----BEGIN RSA PRIVATE KEY-----" \
    "MG8CAQAwEwYHKoZIzj0CAQYIKoZIzj0DAQEEVTBTAgEBBBiYwF7DmgkMs5nuBTGo" \
    "qStzE1qIw4I8oR6hNAMyAARglVCk/eB1iMoPWZO+GyLAdpb7PykKintU3m9cS8cN" \
    "/I3gU7NXutYjL9npMWLoHPI=" \
    "-----END PRIVATE KEY-----"
```

If we wanted to match nothing meaningful before the b64, I might go with:
```
(^[^a-zA-Z0-9+/]*|['"])[a-zA-Z0-9+/]{60,}
```
or something like this.
But even that I am not so sure would not lead to false negatives.

Reply:
That's a good point. I agree that the long base64 string by itself will not lead to many false negatives.